### PR TITLE
Implemented new method to get all available report quantities categories and all report quantities.

### PR DESCRIPTION
**Pull request by maxcapodi78**
Furthermore now the report_by_templates classes supports None expression as argument as they will take all quantities of default category.

Added method to identify if a setup or a sweep is solved or not by checking the solution data.

### CODE REVIEW

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM. Great!

File: pyaedt/modules/SolveSetup.py [hunk 1]
```diff
@@ -1337,13 +1367,29 @@ def __init__(self, app, solutiontype, setupname="MySetupAuto", isnewsetup=True):
...
+                                self.sweeps.append(SweepHFSS3DLayout(self, setupname, el, props=app[el]))
 
                     self.props = SetupProps(self, OrderedDict(setup_data))
             except:
                 self.props = SetupProps(self, OrderedDict())
         self.auto_update = True
 
+    @property
+    def is_solved(self):
+        """Verify if solutions are available for given setup.
```
Comment (MaxJPRey) [inline pyaedt/modules/SolveSetup.py:1379]:
```suggestion
        """Verify if solutions are available for a given setup.
```

File: pyaedt/modules/PostProcessor.py [hunk 2]
```diff
@@ -609,6 +610,113 @@ def __init__(self, app):
...
+        """
+        return list(self.oreportsetup.GetAvailableReportTypes())
+
+    @pyaedt_function_handler()
+    def available_display_types(self, report_category=None):
+        """Retrieve display types for a report categories.
+
+        Parameters
+        ----------
+        report_category : str
```
Comment (MaxJPRey) [inline pyaedt/modules/PostProcessor.py:630]:
```suggestion
        report_category : str, optional
```

**Comment by MaxJPRey** [comment]
LGTM.

**Review by cmesibov — APPROVED** [review]

**Pull request merged**